apps/server/src/services/rundown-service/delayedRundown.utils.ts

```diff
@@ -171,6 +171,41 @@ export async function cachedReorder(eventId: string, from: number, to: number) {
   return reorderedEvent;
 }
 
+/**
+ * Swaps two events
+ * @param {number} fromEventIndex
+ * @param {number} toEventIndex
+ */
+export async function cachedSwap(fromEventIndex: number, toEventIndex: number) {
+  const updatedRundown = DataProvider.getRundown();
+
+  const fromEvent = updatedRundown.at(fromEventIndex);
+  const toEvent = updatedRundown.at(toEventIndex);
+
+  if (!fromEvent || !toEvent) {
+    invalidateFromError();
+  }
+
+  if (fromEvent.type === SupportedEvent.Event && toEvent.type === SupportedEvent.Event) {
+    updatedRundown[fromEventIndex] = {
+      ...toEvent,
+      timeStart: fromEvent.timeStart,
+      timeEnd: fromEvent.timeEnd,
+      duration: fromEvent.duration,
+    };
+
+    updatedRundown[toEventIndex] = {
+      ...fromEvent,
+      timeStart: toEvent.timeStart,
+      timeEnd: toEvent.timeEnd,
+      duration: toEvent.duration,
+    };
+
+    runtimeCacheStore.invalidate(delayedRundownCacheKey);
```

Comment:
I dont think we need to invalidate the entire cache

We know that we would only be swapping event types
We also know that the event types delay is calculated from its position in the rundown

With this in mind swapping the delay values should be safe.
I might be missing something, in which case we can use the `getDelayAt()` which will calculate delay backwards from an index

Reply:
TBH, I'm not so familiar with the implementation here to give a response. I'll keep learning as I go!